arangod/Scheduler/SupervisedScheduler.cpp

```diff
@@ -500,6 +506,44 @@ void SupervisedScheduler::shutdown() {
   }
 }
 
+Result SupervisedScheduler::detachThread() {
+  std::lock_guard<std::mutex> guard(_mutex);
+  // First see if we have already reached the limit:
+  if (_numDetached >= _maxNumberDetachedThreads) {
+    return Result(TRI_ERROR_TOO_MANY_DETACHED_THREADS);
+  }
+
+  // Now we have access to the _workerStates and _detachedWorkerStates
+  // Let's first find ourselves in the _workerStates:
+  uint64_t myNumber = Thread::currentThreadNumber();
+  auto it = _workerStates.begin();
+  while (it != _workerStates.end()) {
+    if ((*it)->_thread->threadNumber() == myNumber) {
+      break;
+    }
+    ++it;
+  }
+  if (it == _workerStates.end()) {
+    return Result(TRI_ERROR_INTERNAL,
+                  "scheduler thread for detaching not found");
+  }
+  std::shared_ptr<WorkerState> state = *it;
+  _workerStates.erase(it);
+  // Since the thread is effectively taken out of the pool, decrease the
+  // number of workers.
+  --_numWorkers;
+  {
+    std::unique_lock<std::mutex> guard(state->_mutex);
+    state->_stop = true;  // We will be stopped after the current task is done
+                          // We know that we are working, so we do not
+                          // have to wake the thread.
+  }
+  ++_metricsThreadsStopped;
```

Comment:
Is it ok to count the thread as `stopped` here already? The thread is still about to perform one more task, which can take an arbitrary amount of time. And until it has finished, the `stopped` counter would be off by one.

Reply:
I think this is correct or at least consistent with prior work. I copy/pasted it from `SupervisedScheduler::StopOneThread` and there it is exactly the same. There it is exactly the same, the thread is still running and working, but already counts as stopped for the purpose of this metric. I would prefer to leave it this way.